selfupdate-docker/test/test-tedge-updater.sh

```diff
@@ -0,0 +1,203 @@
+       --mount type=bind,source=/var/run/docker.sock,target=/run/docker.sock \
+       --mount type=bind,source=/lib/ld-linux-aarch64.so.1,target=/lib/ld-linux-aarch64.so.1 \
+       --mount type=bind,source=/lib/aarch64-linux-gnu,target=/lib/aarch64-linux-gnu \
+       tedge_image:v0.6.1-before-update
+set -
+
+
+# 
+# (4) Simulate selfupdate by directly calling tedge_updater
+#
+
+#
+# (4.a) prepare test
+
+test_info "Providing new image to current container"
+
+# provide new image to existing container (would be done by SM agent in normal flow)
+set -x
+docker cp ./files/tedge_image-v0.6.1-after-update.docker-image-tarx thin-edge:/
+docker exec -it thin-edge sh -c "echo -e 'components = [ "tedge_image" ]' >/etc/tedge/tedge_components.toml"
+set -
+
+
+## decouple hooks for tedge_updater to call them manually for that test
+#test_info "Decoupling hooks to call them manually, for better test-control"
+#set -x
+#docker exec -it thin-edge sh -c "mv /etc/tedge/hooks/hook-start-or-rollback /etc/tedge/hooks/h2.sh"
+#docker exec -it thin-edge sh -c "mv /etc/tedge/hooks/hook-stop-and-snapshot /etc/tedge/hooks/h1.sh"
+#docker exec -it thin-edge sh -c "echo 'exit 0' >/etc/tedge/hooks/hook-start-or-rollback"
+#docker exec -it thin-edge sh -c "echo 'exit 0' >/etc/tedge/hooks/hook-stop-and-snapshot"
+#docker exec -it thin-edge sh -c "chmod +x /etc/tedge/hooks/hook-start-or-rollback"
+#docker exec -it thin-edge sh -c "chmod +x /etc/tedge/hooks/hook-stop-and-snapshot"
+#set -
+
+# connect container to C8Y
+test_info "Connecting to C8Y"
+test_info "C8Y account data are hardcoded here! See both script lines below for tenent!"
+set -x
+docker exec -it thin-edge tedge config set c8y.url christoph.basic.stage.c8y.io
+set -
+if [ "$DO_UPLOAD_CERT" = "YES" ]; then
+   test_info "Line below requires manual password input for your C8Y user. USER IS HARDCODED (name: Christoph)!" "force-wait"
+   set -x
+   docker exec -it thin-edge sudo tedge cert upload c8y --user Christoph
+fi
+set -x
+docker exec -it thin-edge tedge connect c8y
+set -
+
+#
+# (4.b) processing tedge_updater test
+
+test_info "### Processing selfupdate"
+
+## call h1 manually (hook stop)
+#test_info "Calling hook1 (hook stop)"
+#set -x
+#docker exec -it thin-edge /etc/tedge/hooks/h1.sh
+#set -
+    ## check:
+    ### now container shall be named "thin-edge_snapshot"
+    ### tedge shall be disconnected and service stopped
+    ### config is in /tmp/tedge-update-store/
+
+if [ "$DO_CONTAINER_START_FAIL" = "YES" ]; then
+   test_info "Forcing new container start to fail"
+   # remove share folder to force container start fail
+   sudo rm -rf /tmp/tedge-update-store
+fi
+
+# trigger tedge_updater
+test_info "Triggering tedge_updater"
+set -x
+docker exec -it thin-edge sh -c "
+      echo -e 'install\\ttedge_image\\tv0.6.1-after-update\\t/tedge_image-v0.6.1-after-update.docker-image-tarx' | /bin/tedge_updater update-list --plugin-name /etc/tedge/sm-plugins/tedge_docker_plugin.sh"
```

Comment:
Change in tedge-updater expects now plugin-name without full path.

```suggestion
      echo -e 'install\\ttedge_image\\tv0.6.1-after-update\\t/tedge_image-v0.6.1-after-update.docker-image-tarx' | /bin/tedge_updater update-list --plugin-name tedge_docker_plugin.sh"
```